smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -95,88 +121,50 @@ public void generateRequestSerializers(GenerationContext context) {
-                });
-                writer.write("");
-                return null;
-            }
-
-            @Override
-            public Void listShape(ListShape shape) {
-                // TODO Collection cleanup point
-                // There's a decent bit of this "collections are different" work in here, meaning
-                // there's likely to also be that in the implementations. There should be both a
-                // centralized way to check for and/or handle these differences.
-                Shape target = context.getModel().getShapeIndex().getShape(shape.getMember().getTarget()).get();
-                if (target instanceof SimpleShape) {
-                    return null;
-                }
-                generateFunctionSignature(shape, (c, s) -> serializeDocumentCollection(c, s.asListShape().get()));
-                return shape.getMember().accept(this);
-            }
-
-            @Override
-            public Void setShape(SetShape shape) {
-                // TODO See collection cleanup note
-                Shape target = context.getModel().getShapeIndex().getShape(shape.getMember().getTarget()).get();
-                if (target instanceof SimpleShape) {
-                    return null;
-                }
-                generateFunctionSignature(shape, (c, s) -> serializeDocumentCollection(c, s.asSetShape().get()));
-                return shape.getMember().accept(this);
-            }
-
-            @Override
-            public Void mapShape(MapShape shape) {
-                generateFunctionSignature(shape, (c, s) -> serializeDocumentMap(c, s.asMapShape().get()));
-                return shape.getValue().accept(this);
-            }
-
-            @Override
-            public Void structureShape(StructureShape shape) {
-                generateFunctionSignature(shape, (c, s) -> serializeDocumentStructure(c, s.asStructureShape().get()));
-                shape.getAllMembers().values().forEach(member -> member.accept(this));
-                return null;
-            }
-
-            @Override
-            public Void unionShape(UnionShape shape) {
-                generateFunctionSignature(shape, (c, s) -> serializeDocumentUnion(c, s.asUnionShape().get()));
-                shape.getAllMembers().values().forEach(member -> member.accept(this));
-                return null;
-            }
-        }));
+        serializingShapes.forEach(shape -> shape.accept(new ShapeSerializingVisitor(context, true)));
     }
 
     @Override
     public void generateResponseDeserializers(GenerationContext context) {
         TopDownIndex topDownIndex = context.getModel().getKnowledge(TopDownIndex.class);
-        for (OperationShape operation : topDownIndex.getContainedOperations(context.getService())) {
+        OperationIndex operationIndex = context.getModel().getKnowledge(OperationIndex.class);
+        Walker shapeWalker = new Walker(context.getModel().getKnowledge(NeighborProviderIndex.class).getProvider());
```

Comment:
Side note: we should add a constructor to Walker that takes a Model and does this implicitly. I've written this same line a lot myself.